### DISCUSSION TITLE
Is there a way to determine if ProxyInterface "isValid"?

### DISCUSSION THREAD

**Discussion by krab** [Kistler-Group/sdbus-cpp#307] [Q&A] [answered]
For example, Qt DBus has https://doc.qt.io/qt-6/qdbusabstractinterface.html#isValid that makes easier to determine if remote interface on ObjectPath is ready to use or it's does not even exist/valid.

I didn't find same functionality in sdbus-cpp, using generated proxy for udisks2 xml. Only exceptions with a message like this:
`[org.freedesktop.DBus.Error.InvalidArgs] No such interface "org.freedesktop.UDisks2.PartitionTable"` 
wtih exception name `e.getName()` == `org.freedesktop.DBus.Error.InvalidArgs` as std::string, should i just compare that string? There is no "error codes". Or should i call `GetManagedObjects` every time for "just not to fail" how it's done in that https://github.com/Kistler-Group/sdbus-cpp/blob/master/examples/org.freedesktop.DBus.ObjectManager/obj-manager-client.cpp#L68 example?

**Comment by sangelovic** [comment, marked as answer]
The only way to determine if a D-Bus object with given interface does exist is to try to construct the proxy, and either you get the exception you mentioned, or not (in which case such object exists).
